sequencer/src/api.rs

```diff
@@ -3738,6 +3745,276 @@ mod test {
+                    .await
+                    .ok()
+                    .flatten();
+                assert!(amount.is_none(), "amount is not none for block {height}")
+            }
+
+            if leaf.height() == EPOCH_HEIGHT * 2 {
+                break;
+            }
+        }
+
+        drop(membership);
+
+        let mut rewards_map = HashMap::new();
+        let mut total_distributed = U256::ZERO;
+        let mut epoch_rewards = HashMap::<EpochNumber, U256>::new();
+
+        while let Some(leaf) = leaves.next().await {
+            let leaf = leaf.unwrap();
+
+            let header = leaf.header();
+            let distributed = header
+                .total_reward_distributed()
+                .expect("rewards distributed is none");
+
+            let block = leaf.height();
+            tracing::info!("verify rewards for block={block:?}");
+            let membership = coordinator.membership().read().await;
+            let epoch_number =
+                EpochNumber::new(epoch_from_block_number(leaf.height(), EPOCH_HEIGHT));
+
+            let block_reward = membership.block_reward(Some(epoch_number)).unwrap();
+            let leader = membership
+                .leader(leaf.leaf().view_number(), Some(epoch_number))
+                .expect("leader");
+            let leader_eth_address = membership.address(&epoch_number, leader).expect("address");
+
+            drop(membership);
+
+            let validators = client
+                .get::<ValidatorMap>(&format!("node/validators/{epoch_number}"))
+                .send()
+                .await
+                .expect("validators");
+
+            let leader_validator = validators
+                .get(&leader_eth_address)
+                .expect("leader not found");
+
+            let distributor =
+                RewardDistributor::new(leader_validator.clone(), block_reward, distributed.into());
+            // Verify that the sum of delegator stakes equals the validator's total stake.
+            for validator in validators.values() {
+                let delegator_stake_sum: U256 = validator.delegators.values().cloned().sum();
+
+                assert_eq!(delegator_stake_sum, validator.stake);
+            }
+
+            let computed_rewards = distributor.compute_rewards().expect("reward computation");
+
+            // Validate that the leader's commission is within a 10 wei tolerance of the expected value.
+            let total_reward = block_reward.0;
+            let leader_commission_basis_points = U256::from(leader_validator.commission);
+            let calculated_leader_commission_reward = leader_commission_basis_points
+                .checked_mul(total_reward)
+                .context("overflow")?
+                .checked_div(U256::from(COMMISSION_BASIS_POINTS))
+                .context("overflow")?;
+
+            assert!(
+                computed_rewards.leader_commission().0 - calculated_leader_commission_reward
```

Comment:
Can this underflow? In practice because we give the round off error funds to the leader probably not, I guess?

Reply:
yeah, the leader receives commission reward greater than what it should because of round offf error funds, so it won't underflow